spec/facter/cache_manager_spec.rb

```diff
@@ -135,6 +136,32 @@
         )
       end
     end
+
+    context 'with timer' do
+      before do
+        allow(File).to receive(:directory?).and_return(true)
+        allow(fact_groups).to receive(:get_fact_group).and_return(group_name)
+        allow(fact_groups).to receive(:get_group_ttls).and_return(nil)
+        allow(fact_groups).to receive(:get_fact).and_return(nil)
+        allow(File).to receive(:readable?)
+        allow(File).to receive(:mtime).with(cache_file_name).and_return(Time.now)
+        allow(Facter::Util::FileHelper).to receive(:safe_read).with(cache_file_name).and_return(cached_core_fact)
+        allow(Facter::Options).to receive(:[]).with(:cache).and_return(true)
+        allow(Facter::Framework::Benchmarking::Timer).to receive(:measure)
+      end
+
+      it 'returns cached external facts' do
+        allow(fact_groups).to receive(:get_fact).with('os').and_return(nil)
+        allow(fact_groups).to receive(:get_fact).with('my_custom_fact').and_return(nil)
+        allow(fact_groups).to receive(:get_fact).with('ext_file.txt').and_return(external_fact)
+        allow(Facter::Util::FileHelper).to receive(:safe_read).with(File.join(cache_dir, 'ext_file.txt'))
+                                                              .and_return(cached_external_fact)
+        allow(File).to receive(:mtime).with(File.join(cache_dir, 'ext_file.txt')).and_return(Time.now)
+
+        cache_manager.resolve_facts(searched_facts)
```

Comment:
You could add an empty line to make the separation between the Arrange, Act and Assertion parts of the unit test.

Reply:
Updated.